### ISSUE TITLE
Route Transitions with Pose

### ISSUE THREAD

**Issue by MrLoh** [jaredpalmer/after.js#172]
# Feature Request

## Problem Description

I was trying to setup After.js with Custom Route transitions, using Posed, as described in this article: 
https://popmotion.io/pose/learn/route-transitions-react-router/

Unfortunately Pose just works properly, if the `Switch` component can be wrapped directly and is passed a `location` prop explicitly. Which is not possible by default with `After.js`

## Suggested Solutions

I got it working by forking the implementation of the `After` component to allow me to wrap the `Switch` component directly in a transition container (see https://github.com/MrLoh/afterjs-apollo-rnw-styled-pose-example/blob/master/src/AfterWithTransition.js)

This is obviously not a great solution. Maybe it would be possible to pass a `Wrapper` component to the `After` component to prevent this issue. Since the `After` component is only used explicitly on the client, this introduces more complexity though. 

I could imagine, that it would be easier, if the interface for `customRenderer` also explicitly uses the `After` component and requires also setting the `StaticRouter` manually. This would also help in setting up content for all routes like a header outside of the router switch, which currently isn't possible on the server. 

I'd be happy to come up with a PR, but this should be discussed first, as it deals with the architecture of After. 

---

- After.js Version: latest
- Razzle Version: 2.4.0
- React Version: 16.4.2
- OS: macOS 10.13.6
- Node Version: 10.1.0
- Package Manager and Version: yarn 1.6.0
- Demo Repo or Link: https://github.com/MrLoh/afterjs-apollo-rnw-styled-pose-example

**Comment by jaredpalmer** [comment]
Yes. Just like gatsby 2, lack of a layout component makes this impossible. Let’s fix it

**Comment by MrLoh** [comment]
Maybe the easiest way would be to pass a layout component to both `After` and to `render` (just like `document`). I think that should work for both: non switch routes and transitions.

**Comment by MrLoh** [comment]
@jaredpalmer I wipped up a quick PR #173 of how this could work

**Comment by MrLoh** [comment]
There’s a PR in flight for this, so it’s not stale.

**Comment by MrLoh** [comment]
PR pending

**Comment by MiYogurt** [comment]
God, to solve the Layout problem and routing animation problem, I need to modify most of the After. JS files. React is very difficult to insert in After. JS at some level.

I thnik in `customRenderer` function should pass `{route, data}` params. let user can self replace `After.tsx` or After.tsx add some wrap component interface。

**Comment by fa7ad** [comment]
Hey people, did anybody ever figure out a simple solution to this?

**Comment by MrLoh** [comment]
yes, switching to next, haven't regretted it for a second

**Comment by fa7ad** [comment]
That doesn't sound like a solution. Specially with that horrible routing system.
